### PR TITLE
Fix wrong fiat value in transaction confirmation screen

### PR DESCRIPTION
**Pull request by estebanmino**
Fixes RNBW-2154

## What changed (plus any additional context for devs)

We're currently asking for the "native price" once when that view is loaded calling ethereum utils `getPriceOfNativeAssetForNetwork`, so if that price is not ready we're using a wrong default value.

This PR introduces a new selector that depends on the current network to get the generic asset, whe the asset is ready we check for the price, if the price is not ready yet we show a loader on the fiat section of the screen until is ready, which we wait through the use of this new selector.

## PoW (screenshots / screen recordings)

https://recordit.co/RbRnUWyWoP

## Dev checklist for QA: what to test

- fiat price makes sense in all networks
- sometimes I was able to reproduce the real but in real time but not always, it should look like
image

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?

### CODE REVIEW

File: src/screens/TransactionConfirmationScreen.js [hunk 1]
```diff
@@ -204,9 +205,17 @@ export default function TransactionConfirmationScreen() {
   } = routeParams;
   const isMessageRequest = isMessageDisplayType(method);
   const [ready, setReady] = useState(isMessageRequest);
+  const genericNativeAsset = useNativeAssetForNetwork(currentNetwork);
 
   const walletConnector = walletConnectors[peerId];
 
+  const priceOfNativeAsset = useMemo(() => {
+    if (greaterThan(genericNativeAsset?.price?.value, 1)) {
```
Comment (jinchung) [inline src/screens/TransactionConfirmationScreen.js:213]: why 1? what is this trying to achieve?

**Comment by jinchung** [comment]
In the original ticket's video - not only was the mint price off, but so was the displayed gas price ($0.02 in the bottom left corner).

**Comment by estebanmino** [comment]
ooh i missed that that changes everything

**Comment by estebanmino** [comment]
@jinchung https://rainbowhaus.slack.com/archives/CGUDDJYNL/p1644441312077409 confirmed here and left it looking like this when the price is not available, price = 0

image

File: src/screens/TransactionConfirmationScreen.js (hunk 1 above)
Comment (estebanmino) [reply src/screens/TransactionConfirmationScreen.js]: removed!

**Comment by skylarbarrera** [comment]
> @jinchung https://rainbowhaus.slack.com/archives/CGUDDJYNL/p1644441312077409 confirmed here and left it looking like this when the price is not available, price = 0
> 
> image

@estebanmino can we tweak the spinning top padding so its got some space from the divider? also what happens if it never loads?

**Review by skylarbarrera — APPROVED** [review]
👌

**Review by jinchung — APPROVED** [review]
🌮

**Review by ibrahimtaveras00 — APPROVED** [review]
good on both OS's 👍🏽

**Pull request merged**